bigquery/google/cloud/bigquery/job.py

```diff
@@ -2570,21 +2656,57 @@ def from_api_repr(cls, resource):
         return cls(
             name=resource.get('name'),
             entry_id=resource.get('id'),
+            start_ms=resource.get('startMs'),
+            end_ms=resource.get('endMs'),
+            input_stages=resource.get('inputStages', []),
+            parallel_inputs=resource.get('parallelInputs'),
+            completed_parallel_inputs=resource.get('completedParallelInputs'),
+            wait_ms_avg=resource.get('waitMsAvg'),
+            wait_ms_max=resource.get('waitMsMax'),
             wait_ratio_avg=resource.get('waitRatioAvg'),
             wait_ratio_max=resource.get('waitRatioMax'),
+            read_ms_avg=resource.get('readMsAvg'),
+            read_ms_max=resource.get('readMsMax'),
             read_ratio_avg=resource.get('readRatioAvg'),
             read_ratio_max=resource.get('readRatioMax'),
+            compute_ms_avg=resource.get('computeMsAvg'),
+            compute_ms_max=resource.get('computeMsMax'),
             compute_ratio_avg=resource.get('computeRatioAvg'),
             compute_ratio_max=resource.get('computeRatioMax'),
+            write_ms_avg=resource.get('writeMsAvg'),
+            write_ms_max=resource.get('writeMsMax'),
             write_ratio_avg=resource.get('writeRatioAvg'),
             write_ratio_max=resource.get('writeRatioMax'),
             records_read=records_read,
             records_written=records_written,
             status=resource.get('status'),
+            shuffle_output_bytes=resource.get('shuffleOutputBytes'),
+            shuffle_output_bytes_spilled=resource.get(
+                    'shuffleOutputBytesSpilled'),
             steps=[QueryPlanEntryStep.from_api_repr(step)
                    for step in resource.get('steps', ())],
         )
 
+    @property
+    def started(self):
+        """Datetime at which the stage started
+
+        :rtype: ``datetime.datetime``, or ``NoneType``
```

Comment:
This should definitely get updated to Google-style.

Example:
https://github.com/GoogleCloudPlatform/google-cloud-python/blob/ef16bdde527fdc0aff908d8aa957482dbea82158/bigquery/google/cloud/bigquery/table.py#L658